pogom/account.py

```
@@ -98,8 +98,190 @@ def check_login(args, account, api, position, proxy_url):
+        parse_new_timestamp_ms(account, response)
+        time.sleep(random.uniform(.2, .3))
+
+    except Exception as e:
+        log.exception('Login for account %s failed. Exception in ' +
+                      'get_player_profile: %s', account['username'], repr(e))
+
+    try:  # 7 - Check if there are level up rewards to claim.
+        request = api.create_request()
+        request.level_up_rewards()
+        request.check_challenge()
+        request.get_hatched_eggs()
+        request.get_inventory(last_timestamp_ms=account['last_timestamp_ms'])
+        request.check_awarded_badges()
+        request.download_settings(hash=account['remote_config']['hash'])
+        request.get_buddy_walked()
+        response = request.call()
+        parse_new_timestamp_ms(account, response)
+        time.sleep(random.uniform(.45, .7))
+
+    except Exception as e:
+        log.exception('Login for account %s failed. Exception in ' +
+                      'level_up_rewards: %s', account['username'], repr(e))
+
+    try:  # 8 - Register Background Device request.
+        request = api.create_request()
+        request.register_background_device(device_type='apple_watch')
+        request.check_challenge()
+        request.get_hatched_eggs()
+        request.get_inventory(last_timestamp_ms=account['last_timestamp_ms'])
+        request.check_awarded_badges()
+        request.download_settings(hash=account['remote_config']['hash'])
+        request.get_buddy_walked()
+        response = request.call()
+        parse_new_timestamp_ms(account, response)
+
+        time.sleep(random.uniform(.53, 1.1))
+    except Exception as e:
+        log.exception('Login for account %s failed. Exception in ' +
+                      'Background Device: %s', account['username'], repr(e))
+
+    # TODO: # 9 - Make a request to get Shop items.
+
+        log.debug('Login for account %s successful.', account['username'])
+        time.sleep(random.uniform(10, 20))
 
```

All exceptions are catched and does not return an error, it just continue the flow, they should return an error or rethrow.